### PR TITLE
Set user role

### PR DESCRIPTION
**Pull request by Neloop**

### CODE REVIEW

File: app/config/permissions.neon [hunk 1]
```diff
@@ -105,10 +105,17 @@ permissions:
           - update
           - remove
           - addSubgroup
-          - addSupervisor
           - removeSupervisor
       conditions: group.isAdmin
 
+    - allow: true
```
Comment (Teyras) [inline app/config/permissions.neon]: This lets the superadmin make a student supervisor of a group. I'm not sure if that's okay...

File: app/V1Module/security/Policies/UserPermissionPolicy.php [hunk 2]
```diff
@@ -31,4 +31,13 @@ public function isNotExternalAccount(Identity $identity, User $user): bool {
     return !$user->hasExternalAccounts();
   }
 
+  public function isSupervisor(Identity $identity, User $user) {
```
Comment (Teyras) [inline app/V1Module/security/Policies/UserPermissionPolicy.php]: An admin should also be a supervisor.

File: app/V1Module/presenters/UsersPresenter.php [hunk 3]
```diff
@@ -430,4 +431,31 @@ public function actionExercises(string $id) {
     $this->sendSuccessResponse($user->getExercises()->getValues());
   }
 
+  public function checkSetRole(string $id) {
+    $user = $this->users->findOrThrow($id);
+    if (!$this->userAcl->canSetRole($user)) {
```
Comment (Teyras) [inline app/V1Module/presenters/UsersPresenter.php]: Which users can set roles? A user with this privilege can make anybody a superadmin, which is pretty dangerous. Maybe we might want to have a non-admin user who can make supervisors...

File: app/config/permissions.neon (hunk 1 above)
Comment (Neloop) [reply app/config/permissions.neon]: Well, thats true, but according to this we should do the checks on supervisor role in set-role endpoint itself?

File: app/V1Module/security/Policies/UserPermissionPolicy.php (hunk 2 above)
Comment (Neloop) [reply app/V1Module/security/Policies/UserPermissionPolicy.php]: Sorry, do not understand.

File: app/V1Module/presenters/UsersPresenter.php (hunk 3 above)
Comment (Neloop) [reply app/V1Module/presenters/UsersPresenter.php]: Like above, do you suggest that we should do the checks of this in the endpoint?

File: app/V1Module/security/ACLModule.php [hunk 4]
```diff
@@ -6,19 +6,23 @@
...
     $this->authorizator = $authorizator;
+    $this->identity = $identity;
   }
 
   protected abstract function getResourceName();
 
   protected function check($action, $context): bool {
     return $this->authorizator->isAllowed(
-      $this->userStorage->getIdentity(),
+      $this->identity ?? $this->userStorage->getIdentity(),
```
Comment (krulis-martin) [inline app/V1Module/security/ACLModule.php]: Would not it be better to set the identity (from arg or from userStorage) in the constructor?

**Pull request merged**